lib/genesis/helpers.go

```diff
@@ -232,33 +316,257 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) error {
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("errror")
+	}
+	return encode, nil
+}
+
+func generateContractKeyValue(c *contracts, prefixKey string, res map[string]string) error {
+	var (
+		key string
+		err error
+	)
+	// First field of contract is the storage key
+	val := reflect.ValueOf(c)
+	if k := reflect.Indirect(val).Type().Field(0).Name; k == currentSchedule {
+		key, err = generateStorageKey(prefixKey, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	encode, err := scale.Marshal(c)
+	if err != nil {
+		return err
+	}
+	res[key] = common.BytesToHex(encode)
+	return nil
+}
+
+func generateKeyValue(s interface{}, prefixKey string, res map[string]string) error {
+	val := reflect.ValueOf(s)
+	n := reflect.Indirect(val).NumField()
+	for i := 0; i < n; i++ {
+		val := reflect.ValueOf(s)
+		storageKey := reflect.Indirect(val).Type().Field(i).Name
+		if storageKey == phantom { //TODO: figure out what to do with Phantom as its value is null
+			continue
+		}
+		key, err := generateStorageKey(prefixKey, storageKey)
+		if err != nil {
+			return err
+		}
+		value, err := generateStorageValue(s, i)
+		if err != nil {
+			return err
+		}
+		res[key] = common.BytesToHex(value)
+	}
+	return nil
+}
+
 func formatKey(kv *keyValue) (string, error) {
 	switch {
-	case reflect.DeepEqual([]string{"grandpa", "authorities"}, kv.key):
+	case reflect.DeepEqual([]string{"Grandpa", "Authorities"}, kv.key):
 		kb := []byte(`:grandpa_authorities`)
 		return common.BytesToHex(kb), nil
-	case reflect.DeepEqual([]string{"system", "code"}, kv.key):
+	case reflect.DeepEqual([]string{"System", "code"}, kv.key):
 		kb := []byte(`:code`)
 		return common.BytesToHex(kb), nil
 	default:
 		if len(kv.key) < 2 {
 			return "", errors.New("key array less than 2")
 		}
-		prefix, err := common.Twox128Hash([]byte(kv.key[0]))
-		if err != nil {
-			return "", err
-		}
-		keydata, err := common.Twox128Hash([]byte(kv.key[1]))
-		if err != nil {
-			return "", err
+
+		return generateStorageKey(kv.key[0], kv.key[1])
+	}
+}
+
+func generateSessionKeyValue(s *session, prefixKey string, res map[string]string) error {
+	val := reflect.ValueOf(s).Elem()
+	moduleName, err := common.Twox128Hash([]byte(prefixKey))
+	if err != nil {
+		return err
+	}
+
+	storageVal, ok := val.Field(0).Interface().([][]interface{})
+	if !ok {
+		return nil
+	}
+
+	for _, strV := range storageVal {
+		for _, v := range strV {
+			var validatorAccID []byte
+			switch t := v.(type) {
+			case string:
+				var nextKeyHash []byte
+				nextKeyHash, err = common.Twox128Hash([]byte("NextKeys"))
+				if err != nil {
+					return err
+				}
+
+				validatorAccID = crypto.PublicAddressToByteArray(common.Address(t))
+				var accIDHash []byte
+				accIDHash, err = common.Twox64(validatorAccID)
+				if err != nil {
+					return err
+				}
+
+				res[common.BytesToHex(append(append(moduleName, nextKeyHash...), append(accIDHash, validatorAccID...)...))] = common.BytesToHex(validatorAccID)
```

Comment:
spliting those appends will be more readable IMO 
```go
prefix := bytes.Join([][]byte{moduleName, nextKeyHash}, []byte{})
sufix := bytes.Join([][]byte{accIDHash, validatorAccID}, []byte{})

res[common.BytesToHex(prefix, sufix...)] = common.BytesToHex(validatorAccID)
```

Reply:
Done